### PR TITLE
Added contract based lambda function to retrieve the catalyst servers list.

### PR DESCRIPTION
**Pull request by marcosnc**

### CODE REVIEW

File: lambdas/src/apis/contracts/TimeRefreshedDataHolder.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export class TimeRefreshedDataHolder<T> {
+
+    private value: T
+
+    constructor(private readonly provider: ()=>Promise<T>, private readonly refreshTime: number) { }
```
Comment (nchamo) [inline lambdas/src/apis/contracts/TimeRefreshedDataHolder.ts:5]: Super nit. Space before and after `=>`

File: lambdas/src/apis/contracts/DAOListsClient.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export class DAOListContractClient {
+
+  constructor(private readonly contract: DAOListContract) {  }
+
+  async getAllValues(): Promise<Set<string>> {
+    const count = await this.contract.getCount()
+    const values: Set<string> = new Set()
+    for (let i = 1; i <= count; i++) {
```
Comment (nchamo) [inline lambdas/src/apis/contracts/DAOListsClient.ts:14]: Why are we starting with 1? Is this ok?

File: lambdas/src/apis/contracts/DAOCache.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { ServerMetadata } from "decentraland-katalyst-commons/ServerMetadata";
+import { TimeRefreshedDataHolder } from "./TimeRefreshedDataHolder";
+import { DAOContractClient } from "decentraland-katalyst-commons/DAOClient";
+import ms from "ms";
+import { DAOListClient } from "./DAOListsClient";
+
+const REFRESH_TIME: number = ms('15m')
```
Comment (nchamo) [inline lambdas/src/apis/contracts/DAOCache.ts:7]: I think we could event set this higher, like `30m`. These values will rarely change

File: lambdas/src/apis/contracts/DAOListsClient.ts (hunk 2 above)
Comment (marcosnc) [reply lambdas/src/apis/contracts/DAOListsClient.ts:14]: Yes... :( 
The 0 position is a reserved value according to Nacho.

File: lambdas/src/apis/contracts/DAOCache.ts (hunk 3 above)
Comment (marcosnc) [reply lambdas/src/apis/contracts/DAOCache.ts:7]: You're right, I'll set it to 30m.

File: lambdas/src/apis/contracts/DAOListsClient.ts (hunk 2 above)
Comment (marcosnc) [reply lambdas/src/apis/contracts/DAOListsClient.ts]: I changed the code to use the `get`  method in the contract, it starts at 0.

**Pull request merged**